src/streams.ts

```diff
@@ -1,9 +1,20 @@
 import * as vscode from 'vscode';
-import { FioraClient } from './client';
+import { FioraClient, FioraMessageItem } from './client';
 import { get } from 'lodash';
+import { client } from './extension';
+import { URL } from 'url';
 
 const outputChannels: Record<string, vscode.OutputChannel> = {};
 
+function formatMessage(msg: FioraMessageItem) {
+  if (msg.type === 'image' && msg.content.startsWith('//')) {
+    const url = client.serviceUrl ? new URL(client.serviceUrl) : null;
+    const protocol = url?.protocol ?? 'https:';
+    return `[${msg.from.username}]: ${protocol}${msg.content}`;
+  }
+  return `[${msg.from.username}]: ${msg.content}`;
+}
+
 export function openConverseOutput(
   context: vscode.ExtensionContext,
   client: FioraClient,
```

Comment:
Hi, Here has client. Just pass it to `formatMessage` is ok

Dont try to export client to global, its should be private.